types/src/block/output/unlock_condition/timelock.rs

```diff
@@ -44,11 +44,32 @@ fn verify_timestamp<const VERIFY: bool>(timestamp: &u32, _: &()) -> Result<(), E
 pub mod dto {
     use serde::{Deserialize, Serialize};
 
+    use super::*;
+    use crate::block::error::dto::DtoError;
+
     #[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
     pub struct TimelockUnlockConditionDto {
         #[serde(rename = "type")]
         pub kind: u8,
         #[serde(rename = "unixTime")]
         pub timestamp: u32,
     }
+
+    impl From<&TimelockUnlockCondition> for TimelockUnlockConditionDto {
+        fn from(value: &TimelockUnlockCondition) -> Self {
+            TimelockUnlockConditionDto {
+                kind: TimelockUnlockCondition::KIND,
+                timestamp: value.timestamp(),
+            }
+        }
+    }
+
+    impl TryFrom<&TimelockUnlockConditionDto> for TimelockUnlockCondition {
+        type Error = DtoError;
+
+        fn try_from(value: &TimelockUnlockConditionDto) -> Result<TimelockUnlockCondition, DtoError> {
+            Ok(TimelockUnlockCondition::new(value.timestamp)
+                .map_err(|_| DtoError::InvalidField("timelockUnlockCondition"))?)
```

Comment:
Should this be `timestamp` or stay `timelockUnlockCondition`?

Reply:
Hmm let's keep it that way for consistency, doesn't really matter anyway